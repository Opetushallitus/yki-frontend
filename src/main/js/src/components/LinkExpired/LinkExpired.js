import PropTypes from 'prop-types';
import React from 'react';
import { withTranslation } from 'react-i18next';

import YkiImage2 from '../../assets/images/ophYki_image2.png';
import HeadlineContainer from '../HeadlineContainer/HeadlineContainer';
import BackButton from '../Registration/BackButton/BackButton';
import classes from './LinkExpired.module.css';

export const linkExpired = props => {
  const { history, match } = props;

  const key = () => {
    switch (match.path) {
      case '/ilmoittautuminen/vanhentunut': {
        return 'registration.expired.loginlink';
      }
      case '/maksu/vanhentunut': {
        return 'registration.expired.paymentlink';
      }
      default: {
        return 'registration.expired.link';
      }
    }
  };

  return (
    <>
      <main id="main">
        <HeadlineContainer
          headlineTitle={props.t(key())}
          headlineContent={null}
          headlineImage={YkiImage2}
          disableContent={true}
        />
        <div className={classes.Content}>
          <BackButton clicked={() => history && history.push('/')} />
          <p>{props.t(`${key()}.info`)}</p>
        </div>
      </main>
    </>
  );
};

linkExpired.propTypes = {
  match: PropTypes.object.isRequired,
  history: PropTypes.object.isRequired,
};

export default withTranslation()(linkExpired);
