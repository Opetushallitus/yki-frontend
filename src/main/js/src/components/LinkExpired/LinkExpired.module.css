.Content {
  width: auto;
  max-width: 100vw;
  margin: 0 14vw 10vh 14vw;
  min-height: 17vh;
}

.Content h1 {
  font: normal normal bold 28px/24px PT Sans;
  letter-spacing: 0;
  color: hsla(194, 82%, 43%, 1);
  padding-bottom: 2vh;
}

.Content p {
  font: normal normal normal 16px/24px PT Sans;
  letter-spacing: 0;
  color: hsla(194, 91%, 21%, 1);
}
