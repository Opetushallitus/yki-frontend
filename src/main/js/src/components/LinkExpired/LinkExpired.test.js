import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import { linkExpired as LinkExpired } from './LinkExpired';

configure({ adapter: new Adapter() });

describe('<LinkExpired />', () => {
  it('should render link expired page', () => {
    const wrapper = shallow(
      <LinkExpired
        t={key => key}
        match={{ path: '/ilmoittautuminen/vanhentunut' }}
        history={history}
      />,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
