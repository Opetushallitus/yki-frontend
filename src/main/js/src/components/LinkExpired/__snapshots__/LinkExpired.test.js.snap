// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LinkExpired /> should render link expired page 1`] = `
<Fragment>
  <main
    id="main"
  >
    <HeadlineContainer
      disableContent={true}
      headlineContent={null}
      headlineImage="ophYki_image2.png"
      headlineTitle="registration.expired.loginlink"
    />
    <div
      className="Content"
    >
      <backButton
        href="/"
      />
      <p>
        registration.expired.loginlink.info
      </p>
    </div>
  </main>
</Fragment>
`;
